support-frontend/assets/pages/contributions-landing/components/PaymentMethodSelector.tsx

```diff
@@ -112,231 +104,277 @@ const mapDispatchToProps = {
 	loadAmazonPaySdk,
 };
 
-// ----- Render ----- //
-const getPaymentMethodLogo = (paymentMethod: PaymentMethod) => {
-	switch (paymentMethod) {
-		case PayPal:
-			return <SvgPayPalDs />;
-
-		case DirectDebit:
-		case ExistingDirectDebit:
-			return <SvgDirectDebitSymbolDs />;
-
-		case AmazonPay:
-			return <SvgAmazonPayLogoDs />;
-
-		case Sepa:
-			return <SvgSepa />;
-
-		case Stripe:
-		case ExistingCard:
-		default:
-			return <SvgNewCreditCardDs />;
-	}
-};
+function PaymentMethodSelector(props: PaymentMethodSelectorProps) {
+	const paymentMethods: PaymentMethod[] = getValidPaymentMethods(
+		props.contributionType,
+		props.switches,
+		props.countryId,
+		props.countryGroupId,
+	);
 
-const legend = (
-	<div className="secure-transaction">
-		<legend id="payment_method">
-			<ContributionChoicesHeader>Payment Method</ContributionChoicesHeader>
-		</legend>
-		<SecureTransactionIndicator modifierClasses={['middle']} />
-	</div>
-);
-
-const renderLabelAndLogo = (paymentMethod: PaymentMethod) => (
-	<>
-		<div>{getPaymentLabel(paymentMethod)}</div>
-		{getPaymentMethodLogo(paymentMethod)}
-	</>
-);
-
-const renderExistingLabelAndLogo = (
-	existingPaymentMethod: RecentlySignedInExistingPaymentMethod,
-) => (
-	<>
-		<div>{getExistingPaymentMethodLabel(existingPaymentMethod)}</div>
-		{getPaymentMethodLogo(
-			mapExistingPaymentMethodToPaymentMethod(existingPaymentMethod),
-		)}
-	</>
-);
-
-// having to do this nasty cast because Flow sucks and type guards don't work through .filter
-const getFullExistingPaymentMethods = (
-	props: PropTypes,
-): RecentlySignedInExistingPaymentMethod[] =>
-	(props.existingPaymentMethods || []).filter(
-		isUsableExistingPaymentMethod,
-	) as any;
-
-const noPaymentMethodsErrorMessage = (
-	<GeneralErrorMessage
-		classModifiers={['no-valid-payments']}
-		errorHeading="Payment methods are unavailable"
-		errorReason="all_payment_methods_unavailable"
-	/>
-);
-const radioCss = {
-	'& + div': {
-		display: 'flex',
-		width: '100%',
-		margin: 0,
-		justifyContent: 'space-between',
-	},
-	'& + div svg': {
-		width: '36px',
-		height: '24px',
-	},
-	'&:not(:checked) + div svg': {
-		filter: 'grayscale(100%)',
-	},
-};
+	const fullExistingPaymentMethods = getFullExistingPaymentMethods(
+		props.existingPaymentMethods,
+	);
 
-const onPaymentMethodUpdate = (
-	paymentMethod: PaymentMethod,
-	props: PropTypes,
-) => {
-	switch (paymentMethod) {
-		case PayPal:
-			if (!props.payPalHasBegunLoading) {
-				props.loadPayPalExpressSdk(props.contributionType);
-			}
+	const onPaymentMethodUpdate = (paymentMethod: PaymentMethod) => {
```

Comment:
I moved this inside the component so that it gets the `props` through it's closure. I think we would usually define click handlers like this in that way